modules/core/src/main/scala/org/scalasteward/core/data/Version.scala

```diff
@@ -63,12 +64,17 @@ final case class Version(value: String) {
           // Do not select a version with hash if this version contains no hash.
           (v.containsHash && !containsHash) ||
           // Don't select "versions" like %5BWARNING%5D.
-          !v.startsWithLetterOrDigit
+          !v.startsWithLetterOrDigit ||
+          // Don't select "versions" that are only a hash
+          v.isOnlyHash
         }.sorted
       }
       .lastOption
   }
 
+  private def isOnlyHash: Boolean =
+    Rfc5234.hexdig.rep(8).string.filterNot(startsWithDate).parseAll(value).isRight
```

Comment:
Does this only match hashes with exactly 8 digits? If yes, I would vote for also matching longer hashes.